src/devices/Pcd8544/Pcd8544.cs

```diff
@@ -0,0 +1,487 @@
+        /// <param name="resetPin">The reset pin. Use a negative number if you don't want to use it</param>
+        /// <param name="spiDevice">The SPI device.</param>
+        /// <param name="pwmBacklight">The PWM channel for the back light</param>
+        /// <param name="gpioController">The GPIO Controller.</param>
+        /// <param name="shouldDispose">True to dispose the GPIO controller</param>
+        public Pcd8544(int dataCommandPin, int resetPin, SpiDevice spiDevice, PwmChannel? pwmBacklight = null, GpioController? gpioController = null, bool shouldDispose = true)
+        {
+            if (dataCommandPin < 0)
+            {
+                throw new ArgumentException($"{nameof(dataCommandPin)} must be a valid pin number");
+            }
+
+            _dataCommandPin = dataCommandPin;
+            _pwmBacklight = pwmBacklight;
+            _pwmBacklight?.Start();
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+            _shouldDispose = gpioController == null || shouldDispose;
+            _controller = gpioController ?? new();
+            _resetPin = resetPin;
+            if (resetPin > 0)
+            {
+                _controller.OpenPin(resetPin, PinMode.Output);
+                _controller.Write(resetPin, PinValue.Low);
+                // Doc says at least 100 ns
+                Thread.Sleep(1);
+                _controller.Write(resetPin, PinValue.High);
+            }
+
+            _controller.OpenPin(_dataCommandPin, PinMode.Output);
+
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            _bias = 4;
+            _temperature = Temperature.Coefficient0;
+            _contrast = 0x30;
+            _enabled = true;
+            // Extended function, contrast to 0x30, temperature to coef 0, bias to 4, Screen to normal display power on, display to normal mode
+            SpiWrite(false, new byte[] { (byte)(FunctionSet.PowerOn | FunctionSet.ExtendedMode), (byte)(0x80 | _contrast), (byte)Temperature.Coefficient0, (byte)(0x10 | _bias), (byte)FunctionSet.PowerOn, (byte)DisplayControl.NormalMode });
+            Clear();
+            Refresh();
+        }
+
+        #region properties
+
+        /// <summary>
+        /// Enable the screen
+        /// </summary>
+        public bool Enabled
+        {
+            get => _enabled;
+            set
+            {
+                _enabled = value;
+                byte enab = (byte)(_enabled ? FunctionSet.PowerOn : FunctionSet.PowerOff);
+                SpiWrite(false, new byte[] { enab });
+            }
+        }
+
+        /// <summary>
+        /// Change the back light from 0 to 100
+        /// </summary>
+        public float BacklightBrightness
+        {
+            get => _backlightVal;
+            set
+            {
+                if (_pwmBacklight != null)
+                {
+                    _backlightVal = value > 1 ? 1 : value;
```

Comment:
Handle negative values?

Reply:
yes, good catch. I had a byte and change to float  because of the PWM. Didn't adjust. Thanks!